crates/generated_parser/src/ast_builder.rs

```diff
@@ -4592,6 +4675,83 @@ impl<'alloc> AstBuilder<'alloc> {
         self.bindings.truncate(i)
     }
 
+    // Returns the index of the first break or continue at/after `offset` source position.
+    fn find_first_break_or_continue(&mut self, offset: usize) -> BreakOrContinueIndex {
+        let mut i = self.breaks_and_continues.len();
+        for info in self.breaks_and_continues.iter_mut().rev() {
+            if info.offset < offset {
+                break;
+            }
+            i -= 1;
+        }
+        BreakOrContinueIndex::from(i)
+    }
+
+    fn pop_breaks_and_continues_with_label_from_index(
```

Comment:
to match other 2 methods, `pop_labelled_breaks_and_continues_from` or `pop_labelled_breaks_and_continues_from_index` maybe?